src/main/java/org/gridsuite/modification/utils/ModificationUtils.java

```diff
@@ -687,8 +689,36 @@ public ReportNode modifyBranchConnectivityAttributes(ConnectablePosition<?> conn
                                                          ReportNode connectivityReports) {
         List<ReportNode> reports = new ArrayList<>();
         processConnectivityPosition(connectablePosition, connectablePositionAdder, modificationInfos, branch.getNetwork(), reports);
-        modifyConnection(modificationInfos.getTerminal1Connected(), branch, branch.getTerminal1(), reports, ThreeSides.ONE);
-        modifyConnection(modificationInfos.getTerminal2Connected(), branch, branch.getTerminal2(), reports, ThreeSides.TWO);
+
+        List<NetworkModificationException> exceptions = new ArrayList<>();
+        // Pair for Side and Pair for update and terminal
+        List<Pair<ThreeSides, Pair<AttributeModification<Boolean>, Terminal>>> sidePairs =
+                List.of(
+                        Pair.create(ThreeSides.ONE, Pair.create(modificationInfos.getTerminal1Connected(), branch.getTerminal1())),
+                        Pair.create(ThreeSides.TWO, Pair.create(modificationInfos.getTerminal2Connected(), branch.getTerminal2()))
+                );
+        // We want informations for the both sides, not only the first in error
+        for (Pair<ThreeSides, Pair<AttributeModification<Boolean>, Terminal>> side : sidePairs) {
+            try {
+                modifyConnection(side.getSecond().getFirst(), branch, side.getSecond().getSecond(), reports, side.getFirst());
+            } catch (NetworkModificationException nme) {
+                exceptions.add(nme);
+            }
+        }
+        if (exceptions.size() == 1) {
+            // One exception to throw for one side
+            throw exceptions.getFirst();
+        } else if (exceptions.size() > 1) {
+            // One exception to throw for the two sides
+            List<Pair<String, String>> errorPairs = new ArrayList<>(); // Pair for Side and Action
+            errorPairs.add(Pair.create(String.valueOf(ThreeSides.ONE.getNum()), branch.getTerminal1().isConnected() ? DISCONNECT : CONNECT));
+            errorPairs.add(Pair.create(String.valueOf(ThreeSides.TWO.getNum()), branch.getTerminal2().isConnected() ? DISCONNECT : CONNECT));
+            throw new NetworkModificationException(exceptions.getFirst().getType(),
+                    String.format("Could not %s equipment '%s' on side %s",
+                            errorPairs.stream().map(Pair::getSecond).distinct().collect(Collectors.joining("/")),
+                            branch.getId(),
+                            errorPairs.stream().map(Pair::getFirst).collect(Collectors.joining(" & "))));
```

Comment:
I think that you are complicating things here. Won't it always be "1 & 2" anyway ?
```suggestion
                            "1 & 2"))));
```